### DISCUSSION TITLE
Add ability to make it fail if a dependency is secondary level

### DISCUSSION THREAD

**Discussion by TomHart** [composer-unused/composer-unused#613] [Ideas]
Right now if a package requires something another dependency requires, but my code doesn't actually use it, `composer-unused` will still pass, but say `(required by: xyz/some-package)`.

I'd like the ability to pass a flag/config to say "If the package is only required by another package, and not actually used locally, fail"

**Comment by icanhazstring** [comment]
Hi. Actually a good idea.
Was thinking about adding a `--strict` option to be strict about direct dependencies.

**Comment by TomHart** [comment]
Reason this came to mind was I've been working on a big upgrade across many packages, and I need to add dev versions to mid-level packages so composer is happy, and I can easily see myself forgetting to remove them afterwards, so this would help flag that
